llumnix/backends/vllm/migration_backend.py

```diff
@@ -100,31 +95,37 @@ def migrate_cache(self, src_handle, src_blocks: List[int], dst_blocks: List[int]
             ray_obj = self.actor.exec_method.remote(self.is_driver_worker, src_handle, "do_send", None, send_blocks)
             if rpc_numpy_cache is not None:
                 self.do_recv(rpc_numpy_cache, recv_blocks)
-            rpc_numpy_cache = ray.get(ray_obj)
+            rpc_numpy_cache_ref = ray.get(ray_obj)
+            rpc_numpy_cache = ray.get(rpc_numpy_cache_ref)
             recv_blocks = dst_blocks[start_idx:start_idx+offset]
         self.do_recv(rpc_numpy_cache, recv_blocks)
 
     def do_send(self, dst_handle, blocks: List[int]):
         num_blocks = len(blocks)
-        send_cache = self.dummy_cache[:num_blocks].view(self.num_layers, 2, num_blocks, self.migration_cache_size)
+        dummy_cache_idx = self.get_available_cache()
```

Comment:
Can we follow the functions of queue? For example, add non_blocking parameters, and provide blocking/non-blocking functions.

Reply:
The non-blocking feature does not have obvious use at the moment. It is currently sufficient to meet the needs and is very concise.